src/builtin.jq

```diff
@@ -6,6 +7,8 @@ def sort_by(f): _sort_by_impl(map([f]));
 def group_by(f): _group_by_impl(map([f]));
 def unique: group_by(.) | map(.[0]);
 def unique_by(f): group_by(f) | map(.[0]);
+def min(s): reduce s as $x (first(s); if $x < . then $x else . end);
+def max(s): reduce s as $x (first(s); if $x > . then $x else . end);
```

Comment:
Using stream twice causes problem with `input`. For example, `echo 1 | jq -n 'min(input)'` should behave as `echo 1 | jq -n '[input] | min'`, but the former yields break error.

Reply:
If the thunk `s` is side-effect-free, then this is OK, but a) we do have some side-effect-having builtins as @itchyny points out, b) if and when we add an FFI this will become more of a problem.

I suggest: `def min(xs): reduce xs as $x ([]; if length == 0 then [$x] elif $x < .[0] then setpath([0]; $x) else . end) | select(length > 0)[0];`.  We use this `reduce xs as $x ([]; ...)` trick where if the reduction state ends up being the empty array then that means there were no xs (`xs` was `empty`).

BTW, I like the Haskell convention of using one letter for the item and "pluralizing" it to get the name of the stream we're iterating.  `xs` is the plural of "x" (`$x`).  I will be adopting that convention in my jq code.

Reply:
And not evaluating twice to produce the first `$x` also improves performance even for side-effect-free `xs`.  Probably not much in most cases, but hey.